File: php/features.php
<?php
declare(strict_types=1);

// SoftvpnMobile SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class SoftvpnMobileFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new SoftvpnMobileBaseFeature();
            case "test":
                return new SoftvpnMobileTestFeature();
            default:
                return new SoftvpnMobileBaseFeature();
        }
    }
}

File: py/features.py
# SoftvpnMobile SDK feature factory

from feature.base_feature import SoftvpnMobileBaseFeature
from feature.test_feature import SoftvpnMobileTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SoftvpnMobileBaseFeature(),
        "test": lambda: SoftvpnMobileTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
